packages/protocol/engine/tangle/booker/booker.go

```diff
@@ -450,61 +470,102 @@ func (b *Booker) setupEvents() {
 	}, b.workers.CreatePool("Booker", 2))
 }
 
+func (b *Booker) OrphanAttachment(block *Block) {
+	if tx, isTx := block.Transaction(); isTx {
+		attachmentBlock, attachmentOrphaned, lastAttachmentOrphaned := b.attachments.AttachmentOrphaned(tx.ID(), block)
+
+		if attachmentOrphaned {
+			b.Events.AttachmentOrphaned.Trigger(attachmentBlock)
+		}
+
+		if lastAttachmentOrphaned {
+			// TODO: attach this event somewhere to the engine
+			b.Events.Error.Trigger(errors.Errorf("transaction %s orphaned", tx.ID()))
+			b.Ledger.PruneTransaction(tx.ID(), true)
+		}
+	}
+}
+
 // region FORK LOGIC ///////////////////////////////////////////////////////////////////////////////////////////////////
 
 // PropagateForkedConflict propagates the forked ConflictID to the future cone of the attachments of the given Transaction.
 func (b *Booker) PropagateForkedConflict(transactionID, addedConflictID utxo.TransactionID, removedConflictIDs utxo.TransactionIDs) (err error) {
-	for blockWalker := walker.New[*Block]().PushAll(b.attachments.Get(transactionID)...); blockWalker.HasNext(); {
-		block := blockWalker.Next()
+	blockWalker := walker.New[*Block]()
 
-		updated, propagateFurther, forkErr := b.propagateForkedConflict(block, addedConflictID, removedConflictIDs)
-		if forkErr != nil {
-			blockWalker.StopWalk()
-			return errors.Wrapf(forkErr, "failed to propagate forked ConflictID %s to future cone of %s", addedConflictID, block.ID())
-		}
-		if !updated {
-			continue
-		}
+	for it := b.GetAllAttachments(transactionID).Iterator(); it.HasNext(); {
```

Comment:
Add a comment explaining the propagation to weak and liked for the first level of children.